lib/galaxy/config/schemas/config_schema.yml

```diff
@@ -3701,6 +3701,23 @@ mapping:
           Activate this only if you have setup a Celery worker for Galaxy.
           For details, see https://docs.galaxyproject.org/en/master/admin/production.html
 
+      celery_user_rate_limit:
+        type: float
+        default: 0.0
+        required: false
+        desc: |
+          If set to a non-0 value, upper limit on number of
+          tasks that can be executed per user per second.
+
+      celery_user_rate_limit_standard_before_start:
+        type: bool
+        default: false
+        required: false
+        desc: |
+          Applies if celery_user_rate_limit is non-zero. Used for testing against
+          a postgres db. Forces use of standard sql code rather than
```

Comment:
Is this config option used for tests only? If so, can we keep it in the testing code and not in the config schema? (we try to limit our user-facing configuration to options relevant for Galaxy admins). I don't think an admin would need to select what SQL dialect is being used.

Reply:
Originally I did not put it in the config schema for the reason you mention. That is why I referenced in the code using the self.config.get("celery_user_rate_limit_standard_before_start", False) syntax which you commented on earlier. However, it occurred to me that there is a chance, albeit small, that a developer may in the future introduce a config parameter with the same name in the config schema. So I decided to put it in the config schema. Without a central registry of all parameters including those used only for testing there is always a chance of a naming collision. However, I'm ok with removing it from the config schema and using the self.config.get syntax to get its value.

Reply:
Ah, I see! I didn't realize at the time that it was only there for the purpose of setting up the right configuration for a test. Given that this option is only relevant in the context of a test, it (ideally) shouldn't be part of the main code base. (I say "ideally" because we still have a couple of settings like that left over from ancient times, but we try to remove those as time permits). However, you don't really need it: Galaxy can determine what path to take based on the value of `database_connection`: with a non-postgres connection a test will automatically take the standard sql path.

Reply:
I thought it would be useful to test a the standard sql path even with postgres. That is why I introduced this testing parameter. Given that it would have to be in the main code base.

Reply:
It'll never take that path: any such alternative sql implementations in galaxy are strictly postgres OR anything else (with sqlite being the default and mysql no longer formally supported, but know to work). So as long as the conn string is identified as postgres, the postgres-specific implementation should be selected.